### PR TITLE
BPF: Host endpoint support

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Implement host endpoint support for eBPF mode.

* Change the data interface programs to support policy.  Where no policy program is provided, fall back to current behaviour.
* Cooperate between BPF and Linux conntrack to ease upgrade to BPF mode.
* Add support for failsafe ports.
* Implement IPIP/VXLAN policing and bypass.
* Collect the pre-DNAT information about the packet along with Booleans for "to/from host" and pass those to the policy program.
* In the policy program, implement pre-DNAT/AoF and wildcard host endpoint policy. (doNotTrack policy is not yet implemented)
* Upstream various fixes and tweaks from the Enterprise codebase that the HEP work depends on.

## Todos
- [x] Unit tests (full coverage)
- [x] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
eBPF mode now supports host endpoints.  Wildcard host endpoints, pre-DNAT policy and apply on forward policy is supported but doNotTrack policy is not yet supported.  Wildcard host endpoints do not apply to the loopback device in eBPF mode.  This change also adds IPIP and VXLAN bypass and policing and failsafe ports, bringing the eBPF dataplane into line with iptables.
```

### CODE REVIEW

File: bpf/polprog/pol_prog_builder.go
Comment (neiljerram) [inline bpf/polprog/pol_prog_builder.go:155]: Do we need this in OSS?

File: bpf/polprog/pol_prog_builder.go
Comment (neiljerram) [inline bpf/polprog/pol_prog_builder.go:164]: And this?

File: bpf/polprog/pol_prog_builder.go
Comment (neiljerram) [inline bpf/polprog/pol_prog_builder.go:359]: Looks like noProfileMatchID unused.

File: dataplane/linux/bpf_ep_mgr.go
Comment (neiljerram) [inline dataplane/linux/bpf_ep_mgr.go:801]: Would be slightly nicer to put the len check inside extractTiers.

Also I note you're using a subtly different condition here: `len() == 1` vs `len() > 0` above.

**Review by neiljerram — APPROVED** [review]
LGTM.  Just a few pieces of rule match ID that I believe aren't needed in OSS, and so could be removed.  But not a big deal.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Pull request merged**
